src/neuroconv/utils/json_schema.py

```diff
@@ -70,104 +69,93 @@ def get_base_schema(
     return base_schema
 
 
-def get_schema_from_method_signature(method: Callable, exclude: list = None) -> dict:
+def get_schema_from_method_signature(method: Callable, exclude: Optional[List[str]] = None) -> dict:
+    """Deprecated version of `get_json_schema_from_method_signature`."""
+    message = (
+        "The method `get_schema_from_method_signature` is now named `get_json_schema_from_method_signature`."
+        "This method is deprecated and will be removed after January 2025."
+    )
+    warnings.warn(message=message, category=DeprecationWarning, stacklevel=2)
+
+    return get_json_schema_from_method_signature(method=method, exclude=exclude)
+
+
+def get_json_schema_from_method_signature(method: Callable, exclude: Optional[List[str]] = None) -> dict:
     """
-    Take a class method and return a json-schema of the input args.
+    Get the equivalent JSON schema for a signature of a method.
+
+    Also uses `docstring_parser` (NumPy style) to attempt to find descriptions for the arguments.
 
     Parameters
     ----------
-    method: function
-    exclude: list, optional
+    method : callable
+        The method to generate the JSON schema from.
+    exclude : list of str, optional
+        List of arguments to exclude from the schema generation.
+        Always includes 'self' and 'kwargs'.
 
     Returns
     -------
-    dict
-
+    json_schema : dict
+        The JSON schema corresponding to the method signature.
     """
-    if exclude is None:
-        exclude = ["self", "kwargs"]
-    else:
-        exclude = exclude + ["self", "kwargs"]
-    input_schema = get_base_schema()
-    annotation_json_type_map = dict(
-        bool="boolean",
-        str="string",
-        int="number",
-        float="number",
-        dict="object",
-        list="array",
-        tuple="array",
-        FilePathType="string",
-        FolderPathType="string",
+    exclude = exclude or []
+    exclude += ["self", "cls"]
+
+    signature = inspect.signature(obj=method)
+    parameters = signature.parameters
+    additional_properties = False
+    arguments_to_annotations = {}
+    for argument_name in parameters:
+        if argument_name in exclude:
+            continue
+
+        parameter = parameters[argument_name]
+
+        if parameter.kind == inspect.Parameter.VAR_KEYWORD:  # Skip all **{...} usage
+            additional_properties = True
+            continue
+
+        annotation = parameter.annotation
+
+        # Pydantic uses ellipsis for required
+        pydantic_default = ... if parameter.default is inspect._empty else parameter.default
+
+        arguments_to_annotations.update({argument_name: (annotation, pydantic_default)})
+
+    model = pydantic.create_model(
+        "_TempModel", __config__=pydantic.ConfigDict(arbitrary_types_allowed=True), **arguments_to_annotations
     )
-    args_spec = dict()
+
+    temp_json_schema = model.model_json_schema()
+
+    # We never used to include titles in the lower schema layers
```

Comment:
> Does it matter if we do from now on? To avoid adding extra code.

In this PR I am merely trying to maintain as much similarity as possible with the current state

I think we can safely include titles since we artificially generate them on the GUIDE anyway

But maybe this can be step 4 of the chain of PRs

unroot does quite a bit more than just that, so I would rather keep the targetted title cleaning. It's pretty minimal for now

Reply:
OK, thanks for explaining. This makes sense.